src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,1724 @@
+			int inter_Alpha = BLOCK_HEADER_OFFSET_Alpha + MODE_BLOCK_OFFSET_Alpha * ModeID;
+			if (ModeID == 1)
+			{
+				inter_Number = BLOCK_HEADER_OFFSET_Number + MODE1_BLOCK_OFFSET_Number* ModeID;
+			}
+			else
+			{
+				inter_Number = BLOCK_HEADER_OFFSET_Number + MODE1_BLOCK_OFFSET_Number + MODE_BLOCK_OFFSET_Number* (ModeID - 1);
+			}
+			std::ostringstream strs;
+			strs << ModeID;
+
+			int curveID = -1;
+			if (lAlphaBlanks(inter_Alpha - 1)) {
+				ModeName = "Mode" + strs.str();
+			}
+			else
+			{
+				ModeName = Alphas(inter_Alpha - 1);
+			}
+
+			curveID = -1;
+			if (lAlphaBlanks(inter_Alpha)) {
+				InitializeCurve(TEMP_CURVE, curveID);//as this is invalid curve id CalculateCurveVal will return a default when called
+			}
+			else
+			{
+				curveID = GetCurveIndex(Alphas(inter_Alpha));
+				if (curveID == 0) {
+					ShowSevereError("Invalid " + cAlphaFields(inter_Alpha) + '=' + Alphas(inter_Alpha));
+					ShowContinueError("Entered in " + cCurrentModuleObject);
+					ErrorsFound = true;
+					InitializeCurve(TEMP_CURVE, -1);
+				}
+				else { InitializeCurve(TEMP_CURVE, curveID); }
+
+			}
+
+			inter_Alpha = inter_Alpha + 1;
+
+			//A20, \field Mode0 Supply Air Humidity Ratio Lookup Table Name
+			curveID = -1;
+			if (lAlphaBlanks(inter_Alpha)) {
+				InitializeCurve(W_CURVE, curveID);//as this is invalid curve id CalculateCurveVal will return a default  when called
+			}
+			else
+			{
+				curveID = GetCurveIndex(Alphas(inter_Alpha));
+				if (curveID == 0) {
+					ShowSevereError("Invalid " + cAlphaFields(inter_Alpha) + '=' + Alphas(inter_Alpha));
+					ShowContinueError("Entered in " + cCurrentModuleObject);
+					ErrorsFound = true;
+					InitializeCurve(W_CURVE, -1);
+				}
+				else { InitializeCurve(W_CURVE, curveID); }
+
+			}
+			inter_Alpha = inter_Alpha + 1;
+			//A21, \field Mode0 System Electric Power Lookup Table Name
+			curveID = -1;
+			if (lAlphaBlanks(inter_Alpha)) {
+				InitializeCurve(POWER_CURVE, curveID);//as this is invalid curve id CalculateCurveVal will return a default 
+			}
+			else
+			{
+				curveID = GetCurveIndex(Alphas(inter_Alpha));
+				if (curveID == 0) {
+					ShowSevereError("Invalid " + cAlphaFields(inter_Alpha) + '=' + Alphas(inter_Alpha));
+					ShowContinueError("Entered in " + cCurrentModuleObject);
+					ErrorsFound = true;
+					InitializeCurve(POWER_CURVE, -1);
+				}
+				else { InitializeCurve(POWER_CURVE, curveID); }
+
+			}
+			//A22, \field Mode0 Supply Fan Electric Power Lookup Table Name
+			//A23, \field Mode0 External Static Pressure Lookup Table Name
+			//A24, \field Mode0 System Second Fuel Consumption Lookup Table Name
+			//A25, \field Mode0 System Third Fuel Consumption Lookup Table Name
+			//A26, \field Mode0 System Water Use Lookup Table Name
+
+			//N6, \field Mode0  Minimum Outside Air Temperature
+			//N7, \field Mode0  Maximum Outside Air Temperature
+			bool ok = InitializeOutsideAirTemperatureConstraints(Numbers(inter_Number), Numbers(inter_Number + 1)); 
```

Comment:
Are these functions going to be used anywhere else? Are these value checking functions going to be unit tested individually? If yes, then I can see a justification for these functions. Otherwise, it seems like an unnecessary abstraction at the moment.